pir/cpp/client_test.cpp

```diff
@@ -72,12 +77,52 @@ TEST_F(PIRClientTest, TestProcessResponse) {
   Context()->Encoder()->encode(value, pt);
   vector<Ciphertext> ct(1);
   Encryptor()->encrypt(pt, ct[0]);
-  PIRPayload payload = PIRPayload::Load(ct);
 
-  auto result = client_->ProcessResponse(payload).ValueOrDie();
+  Reply reply;
+  *reply.mutable_reply() = SaveCiphertexts(ct).ValueOrDie();
+
+  auto result = client_->ProcessResponse(reply).ValueOrDie();
   ASSERT_EQ(result, value);
 }
 
+TEST_F(PIRClientTest, TestPayloadSerialization) {
+  int64_t value = 987654321;
+  Plaintext pt;
+  Context()->Encoder()->encode(value, pt);
+  vector<Ciphertext> ct(1);
+  Encryptor()->encrypt(pt, ct[0]);
+
+  Reply reply_proto;
+  *reply_proto.mutable_reply() = SaveCiphertexts(ct).ValueOrDie();
+
+  auto reloaded = LoadCiphertexts(Context()->SEALContext(), reply_proto.reply())
+                      .ValueOrDie();
+
+  ASSERT_EQ(reloaded.size(), 1);
+
+  auto keygen_ = make_unique<KeyGenerator>(Context()->SEALContext());
+  auto elts = generate_galois_elts(DEFAULT_POLY_MODULUS_DEGREE);
+  GaloisKeys gal_keys = keygen_->galois_keys_local(elts);
+
+  auto encoded_query = SaveCiphertexts(ct).ValueOrDie();
+  auto encoded_keys = SEALSerialize<GaloisKeys>(gal_keys).ValueOrDie();
+
+  Query proto_query;
+  *proto_query.mutable_query() = encoded_query;
+  proto_query.set_keys(encoded_keys);
```

Comment:
This would be a lot more readable if you create the query proto first and then assign into it directly.